source/fe/fe_poly_tensor.cc

```diff
@@ -1013,10 +1343,173 @@ FE_PolyTensor<POLY,dim,spacedim>::fill_fe_subface_values (
               Assert(false, ExcNotImplemented());
             }
         }
+      // update hessians
+      if (flags & update_hessians )
+
+        {
+          std::vector<Tensor<3, spacedim > > shape_grad_grads1 (n_q_points);
+          std::vector<Tensor<1, dim > > shape_hessians_for_fixed_indices (n_q_points);
+          std::vector<Tensor<1, dim > > transformed_shape_hessians_for_fixed_indices (n_q_points);
```

Comment:
These are just scratch arrays, and you're allocating dynamic memory here every time this function is called. Can you put them as `mutable` fields into the `InternalData` member of this class?